### DISCUSSION TITLE
copier: Renovate does not support git+https

### DISCUSSION THREAD

**Discussion by reitzig** [renovatebot/renovate#31454] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Bitbucket Data Center v8.19.6; Renovate 38.82.1

### Please tell us more about your question or problem

Copier supports source paths with several Git-related protocols:

> If Copier doesn't detect your remote URL as a Git repository, make sure it starts with one of `git+https://`, `git+ssh://`, `git@` or `git://`, or it ends with `.git`.
> https://copier.readthedocs.io/en/stable/generating/

Running Renovate on a repository that contains a `.copier-answers.yaml` with a ` git+https`-URL, we get:

```
git: 'remote-git+https' is not a git command. See 'git --help'.\n
```

I can reproduce a similar error message with my local Git:

```bash
🐧❯ git --version
git version 2.25.1

🐧❯ git clone git+https://bitbucket.some-org/scm/some-project/some-template.git
Cloning into 'some-template'...
fatal: unable to find remote helper for 'git+https'
```

It seems as if Renovate forwarded the URL it found in `.copier-answers.yaml` to `git`; in this case, it would need to remove a `git+` prefix for that to work.

#### Notes

- In our case, plain `https://(...).git` works.
- We would _like_ to use `ssh://` but our Renovate setup uses HTTPS to access our Bitbucket; an automagic mapping would be great (but finicky since Bitbucket HTTPS vs SSH URLs have different structure).

#### References

- [reitzig/reproduction-renovate-31454](https://github.com/reitzig/reproduction-renovate-31454)
- [renovatebot/renovate#29215](https://github.com/renovatebot/renovate/pull/29215/files#diff-03d01a2ea6b7b14347e9a542cc5e5b1ebe808183d78639524716a4de268365d7) -- introduces copier support; tests only extraction of `https`.
- [copier documentation](https://copier.readthedocs.io/en/stable/generating/)

### Logs (if relevant)

<details><summary>Logs</summary>

```
DEBUG: Datasource unknown error (repository=some-project/some-repo, baseBranch=doc/bounded-contexts)
       "datasource": "git-tags",
       "packageName": "git+https://bitbucket.some-org/scm/some-project/some-template.git",
       "err": {
         "task": {
           "commands": [
             "ls-remote",
             "git+https://bitbucket.some-org/scm/some-project/some-template.git"
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "git: 'remote-git+https' is not a git command. See 'git --help'.\n",
         "stack": "Error: git: 'remote-git+https' is not a git command. See 'git --help'.\n\n    at Object.action (/usr/local/renovate/node_modules/.pnpm/simple-git@3.26.0/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:42:29)\n    at PluginStore.exec (/usr/local/renovate/node_modules/.pnpm/simple-git@3.26.0/node_modules/simple-git/src/lib/plugins/plugin-store.ts:54:29)\n    at /usr/local/renovate/node_modules/.pnpm/simple-git@3.26.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:124:42\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/usr/local/renovate/node_modules/.pnpm/simple-git@3.26.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:121:14)\n    at GitExecutorChain.<anonymous> (/usr/local/renovate/node_modules/.pnpm/simple-git@3.26.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:97:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/local/renovate/node_modules/.pnpm/simple-git@3.26.0/node_modules/simple-git/dist/cjs/index.js:52:24)"
       }
DEBUG: Failed to look up git-tags package git+https://bitbucket.some-org/scm/some-project/some-template.git (repository=some-project/some-repo, baseBranch=doc/bounded-contexts, packageFile=test-bc-a/.copier-answers.yml, 

```

</details>

**Comment by rarkins** [comment, marked as answer]
Converted to issue: https://github.com/renovatebot/renovate/issues/31516
